### DISCUSSION TITLE
[Feature]: Example request: Linaria

### DISCUSSION THREAD

**Discussion by knpwrs** [remix-run/remix#2883] [Proposals]
### What is the new or updated feature that you are suggesting?

An example of using Linaria would be excellent. I saw @ionut-botizan mention Linaria here: https://github.com/remix-run/remix/issues/717#issuecomment-991874153

Linaria can be integrated with ESBuild as documented here: https://github.com/callstack/linaria/blob/master/docs/BUNDLERS_INTEGRATION.md#esbuild

### Why should this feature be included?

I think @ionut-botizan's write-up does Linaria justice:

> Linaria is a zero-runtime CSS-in-JS library that compiles your styles into static CSS files at build time, offering a great mix of DX and runtime performance.
> 
> Some of its (cherry-picked) benefits are:
> 
> - Scoped selectors
> - Automatic unused styles removal
> - Automatic vendor prefixing
> - Static CSS (doh) that can be loaded in parallel to JS
> - Works without JavaScript
> - Full power of CSS / No new syntax to learn (you just write CSS in a tagged template literal) with JS variables/expressions interpolation.

My own two cents: I think that Linaria being zero-runtime makes it an excellent fit for Remix.

**Comment by aiji42** [comment]
I have created a sample on how to use linaria in remix. It is safe to wait for the official announcement, but in the meantime I hope this will be helpful.
https://github.com/aiji42/remix-esbuild-override/tree/main/examples/linaria

**Reply by georgiosd** [reply]
Does this mean there's a single render on the server? Yay!

EDIT: I see you have removed it - I'm wondering what your motivation was for that.

**Reply by redbugz** [reply]
It appears it started to break, I'm assuming without knowing how to fix it, it was removed
https://github.com/aiji42/remix-esbuild-override/issues/31

I dug in a bit and found out why the error is occurring and created an issue here:
https://github.com/callstack/linaria/issues/1338

**Comment by ptim** [comment]
I'd love the example to include Critical CSS, if possible!

https://github.com/callstack/linaria/blob/master/docs/CRITICAL_CSS.md
